templates/haproxy.cfg.j2

```diff
@@ -113,8 +118,71 @@ backend tls_{{ vhost.name }}
         {% endfor %}
 
 {% endfor %}
+{% endif %}
+
+{% if 'ocp' in haproxy_type %}
+frontend openshift-app-http
+    {% for bindip in haproxy_ocp_bindips %}
+    bind {{ bindip }}:80
+    {% endfor %}
+    default_backend openshift-app-http-be
+    mode tcp
+    option tcplog
+
+frontend openshift-app-https
+    {% for bindip in haproxy_ocp_bindips %}
+    bind {{ bindip }}:443
+    {% endfor %}
+    default_backend openshift-app-https-be
+    mode tcp
+    option tcplog
+
+frontend openshift-master-api
+    {% for bindip in haproxy_ocp_bindips %}
+    bind {{ bindip }}:6443
+    {% endfor %}
+    default_backend openshift-master-api-be
+    mode tcp
+    option tcplog
+
+
+frontend openshift-master-api-2
+    {% for bindip in haproxy_ocp_bindips %}
+    bind {{ bindip }}:22623
+    {% endfor %}
+    default_backend openshift-master-api-2-be
+    mode tcp
+    option tcplog
+
+backend openshift-app-http-be
+    balance source
+    mode tcp
+    {% for server in haproxy_ocp_app_nodes %}
```

Comment:
for variables/lists like "haproxy_ocp_app_nodes", there is no default in defaults/main.yml but can be easily added.
What about the idea of using a list of ansible group[s] coming from inventory and automatically add entries in haproxy when a node is added in such group ? 
See for example how unbound role does it : 
https://github.com/CentOS/ansible-role-unbound/blob/master/defaults/main.yml#L34
https://github.com/CentOS/ansible-role-unbound/blob/master/templates/02_localzone_cache.conf.j2#L5

That would permit to be "dynamic" on each haproxy role execution (after inventory was just changed) ? Or is that what you had in mind ?